pdc/apps/compose/views.py

```diff
@@ -1364,6 +1366,106 @@ def list(self, request, **kwargs):
         return Response(self._get_composes_for_product_version())
 
 
+class ComposeImageRTTTestViewSet(ChangeSetUpdateModelMixin,
+                                 mixins.ListModelMixin,
+                                 mixins.RetrieveModelMixin,
+                                 StrictQueryParamMixin,
+                                 MultiLookupFieldMixin,
+                                 viewsets.GenericViewSet):
+    """
+    API endpoint that allows querying compose-image RTT Test results.
+
+    ##Test tools##
+
+    You can use ``curl`` in terminal, with -X _method_ (GET|POST|PUT|DELETE),
+    -d _data_ (a json string). or GUI plugins for
+    browsers, such as ``RESTClient``, ``RESTConsole``.
+    """
+    queryset = ComposeImage.objects.select_related('variant_arch', 'image').all()
+    serializer_class = ComposeImageRTTTestSerializer
+    filter_class = ComposeImageRTTTestFilter
+    lookup_fields = (
+        ('variant_arch__variant__compose__compose_id', r'[^/]+'),
+        ('variant_arch__variant__variant_uid', r'[^/]+'),
+        ('variant_arch__arch__name', r'[^/]+'),
+        ('image__file_name', r'[^/]+'),
+    )
+
+    def list(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:composeimagertttests-list$
+
+        __Query params__:
+
+        %(FILTERS)s
+
+        __Response__: a paged list of following objects
+
+        %(SERIALIZER)s
+        """
+        return super(ComposeImageRTTTestViewSet, self).list(*args, **kwargs)
+
+    def retrieve(self, *args, **kwargs):
+        """
+        __Method__: GET
+
+        __URL__: $LINK:composeimagertttests-detail:compose_id}/{variant_uid}/{arch}/{file_name$
+
+        __Response__:
+
+        %(SERIALIZER)s
+        """
+        return super(ComposeImageRTTTestViewSet, self).retrieve(*args, **kwargs)
+
+    def update(self, request, *args, **kwargs):
+        # This method is used by bulk update and partial update, but should not
+        # be called directly.
+        if not kwargs.get('partial', False):
+            return self.http_method_not_allowed(request, *args, **kwargs)
+
+        if not request.data:
+            return NoEmptyPatchMixin.make_response()
+
+        updatable_keys = set(['test_result'])
+        if set(request.data.keys()) - updatable_keys:
```

Comment:
DRF default behavior is to ignore updating on readonly fileds not return a 400 Bad Request.
I know the hard coded set here is hard to trace and maintain, maybe we should find a better way to get rid of all these `updatable_keys` hacks?